prusti-viper/src/encoder/type_encoder.rs

```diff
@@ -392,6 +405,28 @@ impl<'p, 'v, 'r: 'v, 'tcx: 'v> TypeEncoder<'p, 'v, 'tcx> {
                 }
             }
 
+            ty::TyKind::Array(elem_ty, len) => {
+                let array_len =
+                    self.encoder
+                        .const_eval_intlike(&len.val).unwrap()
+                        .to_u64().unwrap();
+                // TODO: use snapshot type at some point
+                // let snapshot = self.encoder.encode_snapshot(&self.ty)?;
+                // let return_ty = snapshot.get_type();
+
+                // need the array's corresponding lookup_pure function as well
+                self.encoder.encode_builtin_function_def(
+                    BuiltinFunctionKind::ArrayLookupPure {
+                        array_ty: vir::Type::TypedRef(format!("{}", elem_ty)),
+                        array_len: array_len.try_into().unwrap(),
+                        return_ty: vir::Type::Int,
+                    });
```

Comment:
This shouldn't be needed. `encode_builtin_function_use` is supposed to lazily call `encode_builtin_function_def` when needed.

Reply:
agree, will remove